army.cpp

```diff
@@ -1261,7 +1261,7 @@ int Army::DoAnAttack(Battle * b, char const *special, int numAttacks, int attack
 		if (tar->riding != -1) attackLevel += mountBonus;
 
 		// 4.4 Check for weapon inflicted bonuses
-		if (attacker->weapon && tar->weapon) {
+		if (attacker->weapon >= 0 && tar->weapon >= 0) {
```

Comment:
what does this change?

Reply:
I have not tested battles with monsters, and this was a bug.

When there is no weapon then the value is `-1` as `0` and up actually means index in weapon table.